src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java

```diff
@@ -29,92 +31,96 @@ public void setUp() {
 		voteId = adminInfo.voteId;
 		adminKey = adminInfo.adminKey;
 	}
-	
+
 	@Test
 	public void the_name_attribute_contains_the_name_of_the_vote() {
 		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		
+
 		assertEquals(result.get("name"), vote.name);
 	}
-	
+
 	@Test
 	public void the_canAddIn_attribute_contains_whether_the_voters_can_add_choices_to_the_vote() {
 		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		
-		assertEquals(result.get("canAddIn"), vote.canAddin);
+
+		assertEquals(result.getJSONObject("voteParameters").get("canAddin"), vote.voteParameters.canAddin);
 	}
 
 	@Test
 	public void the_creationTime_attribute_contains_the_creation_time_of_the_vote() {
 		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		
-		assertEquals(result.get("creationTime"), vote.creationTime);
+
+		assertEquals(result.getLong("creationTime"), vote.creationTime);
 	}
-	
+
 	@Test
 	public void the_choices_attribute_contains_the_choices_of_the_vote() {
 		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		JSONArray jsonArray = vote.createChoicesJson(vote.choices);
-		
-		assertEquals(result.getJSONArray("choices").toString(), jsonArray.toString());
+		String expected = String
```

Comment:
It seems to me that we might be testing implementation and not behaviour here.
For example a string with some spaces and newlines here and there would be as legit a json as the current one and would fail the test.

Reply:
True. BTW, the former implementation generated the expected result with the tested method :)

Reply:
Then the former implementation was not good either. What about testing for a specific attribute value within the json?